src/internal/minikubetestenv/deploy.go

```diff
@@ -22,36 +25,69 @@ import (
 )
 
 const (
-	helmChartLocalPath     = "../../../etc/helm/pachyderm"
 	helmChartPublishedPath = "pach/pachyderm"
-	helmRelease            = "test-release"
-	ns                     = "default"
 	localImage             = "local"
 	licenseKeySecretName   = "enterprise-license-key-secret"
 )
 
-func getPachAddress(t testing.TB) *grpcutil.PachdAddress {
-	cfg, err := config.Read(false, false)
+type DeployOpts struct {
+	Version      string
+	Enterprise   bool
+	AuthUser     string
+	CleanupAfter bool
+	// Because NodePorts are cluster-wide, we use a PortOffset to
+	// assign separate ports per deployment.
+	// NOTE: it might make more sense to declare port instead of offset
+	PortOffset uint16
+}
+
+type helmPutE func(t terraTest.TestingT, options *helm.Options, chart string, releaseName string) error
+
+func helmChartLocalPath(t testing.TB) string {
+	dir, err := os.Getwd()
 	require.NoError(t, err)
+	cnt := 0
+	parts := strings.Split(dir, "/")
+	for i := len(parts) - 1; i >= 0; i-- {
+		cnt++
+		if parts[i] == "src" {
+			break
+		}
+	}
+	return strings.Repeat("../", cnt) + "etc/helm/pachyderm"
+}
 
+func getPachAddress(t testing.TB) *grpcutil.PachdAddress {
+	cfg, err := config.Read(true, true)
+	require.NoError(t, err)
 	_, context, err := cfg.ActiveContext(true)
 	require.NoError(t, err)
-
-	pa, err := grpcutil.ParsePachdAddress(context.PachdAddress)
+	address, err := client.GetUserMachineAddr(context)
 	require.NoError(t, err)
-
-	return pa
+	if address == nil {
+		address = &grpcutil.DefaultPachdAddress
+	}
+	return address
 }
 
 func localDeploymentWithMinioOptions(namespace, image string) *helm.Options {
+	os := runtime.GOOS
+	serviceType := ""
+	switch os {
+	case "darwin":
+		serviceType = "LoadBalancer"
```

Comment:
while working on this I learned that the preferred way to access kube services over localhost on mac is by declaring them as type: `LoadBalancer`, and running `minikube tunnel` in a separate shell session. This works much better than port-forward

Reply:
Death to port-forward! 🔪